In [33]:
# enable many obj dumps in a cell w/out print
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = 'all'

In [35]:
import calendar
import scipy.stats

def print_max(product, data, label=None) :
    
    max_month = max(data, key=data.get)
    
    print("-"*50)
    print(f"Max {product}")
    print("-"*50)
    
    print(f"The month with the highest {'value' if label == None else label} is {max_month}.")
    print(f"  The value was {data[max_month]:,}.\n")

def print_min(product, data, label=None) :

    min_month = min(data, key=data.get)
    
    print("-"*50)
    print(f"Min {product}")
    print("-"*50)
    
    print(f"The month with the lowest {'value' if label == None else label} is {min_month}.")
    print(f"  The value was {data[min_month]:,}.\n")


# List all months
months = []
for month_num in range(1, 13):
    months.append(calendar.month_name[month_num])

# Temperatures (°F)
temperature = [40, 45, 55, 65, 75, 85, 90, 88, 86, 80, 70, 60]

# GreenTote sales
green_tote = [
    87500, 100625, 115725, 132075, 148875, 164500,
    172725, 185800, 180900, 170000, 165000, 160000
]

# SolarTrek Water Bottle sales
solar_trek_water_bottle = [
    80900, 85200, 92500, 100980, 110500, 118300,
    124800, 130500, 128800, 120000, 114000, 110000
]


scipy.stats.pearsonr(green_tote, solar_trek_water_bottle)
scipy.stats.pearsonr(temperature, green_tote)
scipy.stats.pearsonr(temperature, solar_trek_water_bottle)

temperature = dict(zip(months, temperature))
green_tote = dict(zip(months, green_tote))
solar_trek_water_bottle = dict(zip(months, solar_trek_water_bottle))

#print(temperature)
#print(green_tote)
#print(solar_trek_water_bottle)

print_max("GreenTote", green_tote, "sales")
print_min("GreenTote", green_tote, "sales")

print_max("SolarTrek Water Bottle", solar_trek_water_bottle, "sales")
print_min("SolarTrek Water Bottle", solar_trek_water_bottle, "sales")

print_max("Temperature", temperature, "temperature")
print_min("Temperature", temperature, "temperature")

scipy.stats.pearsonr(list(green_tote.values()), list(solar_trek_water_bottle.values()))
scipy.stats.pearsonr(list(temperature.values()), list(green_tote.values()))
scipy.stats.pearsonr(list(temperature.values()), list(solar_trek_water_bottle.values()))


PearsonRResult(statistic=np.float64(0.9880734176209012), pvalue=np.float64(1.8628514007033096e-09))

PearsonRResult(statistic=np.float64(0.9249120857886272), pvalue=np.float64(1.6556171942382135e-05))

PearsonRResult(statistic=np.float64(0.9610765652591183), pvalue=np.float64(6.590554270244513e-07))

--------------------------------------------------
Max GreenTote
--------------------------------------------------
The month with the highest sales is August.
  The value was 185,800.

--------------------------------------------------
Min GreenTote
--------------------------------------------------
The month with the lowest sales is January.
  The value was 87,500.

--------------------------------------------------
Max SolarTrek Water Bottle
--------------------------------------------------
The month with the highest sales is August.
  The value was 130,500.

--------------------------------------------------
Min SolarTrek Water Bottle
--------------------------------------------------
The month with the lowest sales is January.
  The value was 80,900.

--------------------------------------------------
Max Temperature
--------------------------------------------------
The month with the highest temperature is July.
  The value was 90.

---------------------------------------------

PearsonRResult(statistic=np.float64(0.9880734176209012), pvalue=np.float64(1.8628514007033096e-09))

PearsonRResult(statistic=np.float64(0.9249120857886272), pvalue=np.float64(1.6556171942382135e-05))

PearsonRResult(statistic=np.float64(0.9610765652591183), pvalue=np.float64(6.590554270244513e-07))

### 📊 Correlation Analysis & Recommendation

#### 1. GreenTote vs SolarTrek
- **Correlation coefficient (r)**: 0.988  
- **p-value**: 1.86e-09 ✅ **Statistically significant**

➡️ **Interpretation**: There is an **extremely strong positive linear correlation** between GreenTote and SolarTrek sales. The p-value is far below 0.05, meaning this relationship is **not due to random chance**.

#### 2. Temperature vs GreenTote
- **Correlation coefficient (r)**: 0.925  
- **p-value**: 1.66e-05 ✅ **Statistically significant**

➡️ **Interpretation**: GreenTote sales are also **strongly positively correlated with temperature**. Warmer months tend to see higher GreenTote sales, and the low p-value confirms the result is statistically significant.

#### 3. Temperature vs SolarTrek
- **Correlation coefficient (r)**: 0.961  
- **p-value**: 6.59e-07 ✅ **Statistically significant**

➡️ **Interpretation**: SolarTrek sales are **very strongly correlated with temperature** as well, possibly even more than GreenTote. The low p-value confirms this is **highly statistically reliable**.

---

### ✅ Final Recommendation

**📌 Option 1: Develop a joint marketing campaign for the two products.**

> **Justification**:
> - The sales of **GreenTote and SolarTrek are tightly linked** (r = 0.988), meaning customers who buy one are very likely to buy the other.
> - **Both products are influenced by temperature** (r = 0.925 and 0.961), suggesting a **seasonal marketing approach** would be effective.
> - All correlations have **very low p-values (< 0.0001)**, confirming the results are **statistically significant** and not due to random fluctuations.

---

### 🧠 Consider for Future Analysis

Although Option 1 is the clear choice based on data, keep Option 3 in mind for the future:
- Investigate whether **external factors** like **demographics**, **location**, or **promotions** might explain additional variance.
- Collect and test **causal variables**, not just correlated ones.